airflow/utils/db.py

```diff
@@ -1660,11 +1660,14 @@ def upgradedb(
     if errors_seen:
         exit(1)
 
-    if not to_revision and not _get_current_revision(session=session) and not use_migration_files:
+    current_revision = _get_current_revision(session=session)
```

Comment:
Should we move it to inside "with create global lock? While it causes the lock to be applied for all such migrations - for a moment even if current revision is "current", it will have the other migrations to do even less work.

Reply:
That would imply also moving `initdb` into the block. I don’t have context for that—would it work?

Reply:
Agree. Likely not good idea.

But sctually I think there is a problem with current solution. I think it does not work as intended - beuase when migration actually happens in another process, previous_revision will still be as it was when we started.

I think we will need to to just add:

```
previous_revision = current_revision = _get_current_revision(session=session)
```
right after qcquiring the lock.

At this point the database might already be migrated -  if the migration happened in another processs - so both "previous revision" and "current revision" will be already different than before the lock.